firmware/gps/main.cpp

```diff
@@ -0,0 +1,42 @@
+#include "mbed.h"
+#include "TinyGPSPlus.h"
+#include "ArduinoJSON.h"
+
+Serial pc(USBTX,USBRX);             // To my computer
+Serial gps_serial(PA_9, PA_10);     // To the GPS
+TinyGPSPlus GPS;                    // Gps object
+
+float_t lat;
+float_t lon;
+
+char ret[256];
+
+int main()
+{
+    gps_serial.baud(9600);
+    
+    StaticJsonDocument<256> doc;
+    
+    while(1) {
+        if (GPS.encode(gps_serial.getc())) {
+            
+            if (GPS.location.isValid()) {
+                lat = GPS.location.lat();
+                lon = GPS.location.lng(); 
+            }
+            else {
+                lat = -1;
+                lon = -1;
+            }
+
+            doc['latitude'] = lat;
+            doc['longitude'] = lon;
+                
+            serializeJson(doc, ret);
```

Comment:
-1.0, -1.0 is a valid GPS coordinate. It looks like you are expecting `hasSignal` in the json packet in the python code but it is not actually sent here.

Reply:
Uploading new firmware code